### PR TITLE
Pull contributors from GitHub api

### PR DESCRIPTION
**Pull request by ethanzitting**

### CODE REVIEW

File: src/components/Blocks/ProjectDetails/ProjectDetails.tsx [hunk 1]
```diff
@@ -62,18 +54,27 @@ export function ProjectDetails({ links, toolsUsed, contributors }: IProjectDetai
...
+                                                alt={`The github profile picture for ${login}`}
+                                            />
+                                        </div>
+                                        <h3 className="p">{login}</h3>
+                                    </SmartLink>
+                                </li>
+                            ))}
+                        </ul>
+                    ) : (
+                        <div>There was an error loading the contributors</div>
```
Comment (fredlawl) [inline src/components/Blocks/ProjectDetails/ProjectDetails.tsx:76]: How about we don't show there is an error to a user? I'd think a user probably wouldn't care/notice if the list isn't showing, but may raise an eyebrow if they see an error happened.

File: src/pages/projects/[index].tsx [hunk 2]
```diff
@@ -1,197 +1,65 @@
...
-    const router = useRouter();
 
+export const getServerSideProps: GetServerSideProps = async ({ query: { index: url } }) => {
+    let contributors;
     let projectData;
 
-    if (router.asPath.includes('mo-jobs')) {
+    if (url === 'mo-jobs') {
+        const apiContributors = await getProjectContributors('portal-to-work-api');
+        const clientContributors = await getProjectContributors('portal-to-work-client');
```
Comment (fredlawl) [inline src/pages/projects/[index].tsx:20]: Hard coding this is not great. There really no way to map a url to content in such a way this function doesn't need to be updated forevery project we add?

File: src/pages/projects/[index].tsx [hunk 3]
```diff
@@ -1,197 +1,65 @@
...
+
+        contributors = mergeContributors(apiContributors, clientContributors);
```
Comment (fredlawl) [inline src/pages/projects/[index].tsx:22]: In the cases of projects with multiple repos, how about we have a list of repo URL's in the incoming data so then we know that regardless if we have a project with 1..N repos, we'll either merge or not all the contributors together. Ideally we setup the concept of github projects for this case, and I'm not sure if github api will automatically know for the project in total or still only count individual repositories. Either way, this function only accepts up to two, but what if there's 3?

File: src/pages/projects/[index].tsx (hunk 2 above)
Comment (fredlawl) [reply src/pages/projects/[index].tsx:20]: The nice thing about async functions is that they can be chained together to be called at the same time. Here we have two back-to-back awaits which means that before the second await gets called, we have to wait for the first one. These two async calls are mutually exclusive, and therefore can be called in parallel: https://stackoverflow.com/a/35612484

File: src/utils/projectData.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        },
+        {
+            label: 'Ionic',
+            image: {
+                src: 'https://via.placeholder.com/70.png',
+                alt: 'Example Alt Text',
+            },
+        },
+    ],
+    contributors: [
```
Comment (fredlawl) [inline src/utils/projectData.ts:62]: Is this needed anymore?

File: src/utils/projectData.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        },
+        {
+            label: 'MySQL',
+            image: {
+                src: 'https://via.placeholder.com/70.png',
+                alt: 'Example Alt Text',
+            },
+        },
+    ],
+    contributors: [
```
Comment (fredlawl) [inline src/utils/projectData.ts:131]: Is this needed anymore?

**Review by fredlawl — CHANGES_REQUESTED** [review]

File: src/components/Blocks/ProjectDetails/ProjectDetails.tsx (hunk 1 above)
Comment (ethanzitting) [reply src/components/Blocks/ProjectDetails/ProjectDetails.tsx:76]: Good point, I'll just trim the whole contributors section if the api call fails.

File: src/utils/projectData.ts (hunk 4 above)
Comment (ethanzitting) [reply src/utils/projectData.ts]: No, good catch! removed.

**Pull request merged**